patrickstar/core/torch_profiler_hook.py

```diff
@@ -39,13 +39,44 @@
 from patrickstar.utils import get_sys_memory_used, logger
 
 from patrickstar.profiler import profiler
+from patrickstar.utils.memory_monitor import AsyncMemoryMonitor
 
 
-def _update_global_var():
-    gpu_mem_used = get_sys_memory_used(
-        torch.device(f"cuda:{torch.cuda.current_device()}")
+def _cur_mem_usage():
+    """
+    A function used to sample memory usage at the moment
+    before and after an operator sharting and finishinig.
+    """
+    dev = torch.device(f"cuda:{torch.cuda.current_device()}")
+    gpu_mem_used = get_sys_memory_used(dev)
+    return gpu_mem_used
+
+
+def _max_mem_usage_period():
+    """
+    A function used to find the max memory usage during a period time by sampling,
+    between now and the perivous call of this function.
+    ret:
+        the max GPU memory used during this period
+    """
+    monitor = AsyncMemoryMonitor(10)
+    max_mem = monitor.finish()
+    monitor.start()
+    return max_mem
+
+
+def _record_mem_stats():
+    """
+    Record memory statistics at this moment for the profiler.
+    """
+    mem_cur_mon = _cur_mem_usage()
+    # In case of an operator running too short,
+    # the sampler dose not capture any information.
+    # we add mem of cur mom as the default memory usage of the period.
```

Comment:
In case an operator's running time is too short.
dose => dose
cur mom => cur mem